overlord/snapstate/autorefresh_gating.go

```diff
@@ -375,14 +421,6 @@ func heldSnaps(st *state.State) (map[string]bool, error) {
 	held := make(map[string]bool)
 Loop:
 	for heldSnap, holdingSnaps := range gating {
-		refreshed, err := lastRefreshed(st, heldSnap)
-		if err != nil {
-			return nil, err
-		}
-		// make sure we don't hold any snap for more than maxPostponement
-		if refreshed.Add(maxPostponement).Before(now) {
-			continue
-		}
```

Comment:
don't we still need to move this below instead and apply when the holdingSnap is not "system"?

Reply:
Yes, although I think in practice we already enforce on write. I fixed it and added a unit test